lisp/lisplib/libjson/sortedmap.go

```diff
@@ -0,0 +1,92 @@
+package libjson
```

Comment:
stupid question, but is `key?` not needed in the interface here?

Reply:
There is no direct equivalent to the lisp function `key?`. The Go interface uses a Get with two return values `Get(k *LVal) (*LVal, ok)`. So the go interface matches go map semantics, you can't check for the presence of a key without also looking up the value associated with the key.

I think in the future `key?` would probably be implemented through `get`. The only issue is that lisp doesn't directly support multiple return types but if `get` had an optional continuation argument then you can compensate for that.

```lisp
(defun get (m k &optional continuation) ...)

(set 'm (sorted-map "a" 123))

(defun key? (m k) (get m k (lambda (x ok) (if ok true false))))

(defun get-default (m k def) (get m k (lambda (x ok) (if ok x def))))
```